### DISCUSSION TITLE
Perl module .pm doesn't return true value

### DISCUSSION THREAD

**Discussion by Prabal1998** [openssl/openssl#25686] [Q&A]
![image](https://github.com/user-attachments/assets/6270d8c1-59b9-4a92-943c-be1aded585a3)

**Comment by tom-cosgrove-arm** [comment]
`builddata.pm` is generated by `util/mkinstallvars.pl`, which has

```
print <<_____;
our \$VERSION                    = '$values{VERSION}->[0]';
our \@LDLIBS                     =
    # Unix and Windows use space separation, VMS uses comma separation
    \$^O eq 'VMS'
    ? split(/ *, */, '$values{LDLIBS}->[0]')
    : split(/ +/, '$values{LDLIBS}->[0]');

1;
_____
```

and after a clean build mine indeed ends with

```
our @LDLIBS             =
    # Unix and Windows use space separation, VMS uses comma separation
    split(/ +| *, */, ' ');

1;
```

Without knowing how you tried to build, what commands you issued, on what operating system, with which version of OpenSSL, and how you got the code, etc, there is no way to know what's wrong on your system

**Reply by Prabal1998** [reply]
Thank you for your feedback. 
In response to your request--proper installation and logging displays following errors.

**Sequences and Output**

**1.Version** 
Linux OS running on VMworkstation

![image](https://github.com/user-attachments/assets/4252405b-7ed0-4c38-9510-abe2266366bf)
**2.  Openssl version**
![image](https://github.com/user-attachments/assets/2b545c50-73a6-4871-add3-00694d5dea7c)
**3. Installation directory and library path locator**
![image](https://github.com/user-attachments/assets/68b73bcf-5b32-43b4-b6b7-66c3c87768ab)
**4. Compilation logs**
![image](https://github.com/user-attachments/assets/badc905c-f7bb-4881-85e8-3b099f45e9f8)
**5. Installation standard output logs**
![image](https://github.com/user-attachments/assets/bff40069-14b0-4d6c-9778-1855b6437928)

**Comment by tom-cosgrove-arm** [comment]
So there is a [known issue (25366) that OpenSSL 3.3.2 build fails on RHEL 7](https://github.com/openssl/openssl/issues/25366). This is because the version of `List::Util` provided by RHEL 7 is too old.

This was fixed in #25367 and will be released in v3.4.0. If you want to try you might be able to apply the changes from https://github.com/openssl/openssl/pull/25367/files to your local copy to get it to build. (Download `https://github.com/openssl/openssl/pull/25367.diff` and use `git apply`)

**Reply by Prabal1998** [reply]
With appreciation, 

Closing this issue/ticket.
